### PR TITLE
[Distributed] Fix standalone

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
Previous PRs breaks the standalone mode.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/graphsage/experimental/partition_graph.py [hunk 1]
```diff
@@ -45,7 +46,11 @@
...
         g = sym_g
 
+    profiler = Profiler()
+    profiler.start()
     dgl.distributed.partition_graph(g, args.dataset, args.num_parts, 'data',
                                     part_method=args.part_method,
                                     balance_ntypes=balance_ntypes,
                                     balance_edges=args.balance_edges)
+    profiler.stop()
+    print(profiler.output_text(unicode=True, color=True))
```
Comment (VoVAllen) [inline examples/pytorch/graphsage/experimental/partition_graph.py:56]: Is this still needed?

File: python/dgl/distributed/dist_context.py [hunk 2]
```diff
@@ -60,20 +63,25 @@ def initialize(ip_config, num_workers=0, max_queue_size=MAX_QUEUE_SIZE, net_type
     ctx = mp.get_context("spawn")
     global SAMPLER_POOL
     global NUM_SAMPLER_WORKERS
-    if num_workers > 0:
-        SAMPLER_POOL = ctx.Pool(
-            num_workers, initializer=_init_rpc, initargs=(ip_config, max_queue_size,
-                                                          net_type, 'sampler', num_worker_threads))
+    is_standalone = os.environ.get('DGL_DIST_MODE', 'standalone') == 'standalone'
```
Comment (VoVAllen) [inline python/dgl/distributed/dist_context.py:66]: By default it's standalone mode?

File: python/dgl/distributed/dist_context.py [hunk 3]
```diff
@@ -60,20 +63,25 @@ def initialize(ip_config, num_workers=0, max_queue_size=MAX_QUEUE_SIZE, net_type
...
+        connect_to_server(ip_config, max_queue_size, net_type)
     init_role('default')
     init_kvstore(ip_config, 'default')
 
 
 def finalize_client():
     """Release resources of this client."""
-    rpc.finalize_sender()
-    rpc.finalize_receiver()
+    if  os.environ.get('DGL_DIST_MODE', 'standalone') != 'standalone':
```
Comment (VoVAllen) [inline python/dgl/distributed/dist_context.py:82]: Probably make an `is_standalone` function?

File: python/dgl/distributed/dist_graph.py [hunk 4]
```diff
@@ -369,7 +369,7 @@ def _init(self):
         self._client.map_shared_data(self._gpb)
 
     def __getstate__(self):
-        return self.ip_config, self.graph_name, self._gpb_input
+        return self.ip_config, self.graph_name, self._gpb
```
Comment (VoVAllen) [inline python/dgl/distributed/dist_graph.py:372]: Is any change needed in the `__setstate__` function or `_init` function for this?

File: python/dgl/distributed/dist_context.py (hunk 2 above)
Comment (zheng-da) [reply python/dgl/distributed/dist_context.py:66]: yes, it needs to run in Jupyter Notebook by default. The "distributed" mode is turned on by the launch script.

File: python/dgl/distributed/dist_graph.py (hunk 4 above)
Comment (zheng-da) [reply python/dgl/distributed/dist_graph.py:372]: it doesn't. previously, `_gpb_input` may just be None. `_gpb` contains the real graph partition book.

**Pull request merged**